sdks/java/core/src/main/java/org/apache/beam/sdk/util/PubsubClient.java

```diff
@@ -310,16 +317,14 @@ public boolean equals(Object o) {
 
       OutgoingMessage that = (OutgoingMessage) o;
 
-      if (timestampMsSinceEpoch != that.timestampMsSinceEpoch) {
-        return false;
-      }
-      return Arrays.equals(elementBytes, that.elementBytes);
-
+      return Arrays.equals(elementBytes, that.elementBytes)
+             && timestampMsSinceEpoch == that.timestampMsSinceEpoch
+             && Objects.equal(recordId, that.recordId);
```

Comment:
is it worth ordering these in order of fastest comparison? (presumably timestamp first, elements last)?